hydromt_wflow/wflow.py

```diff
@@ -285,6 +293,8 @@ def setup_basemaps(
         # Rename and add to grid
         rmdict = {k: v for k, v in self._MAPS.items() if k in ds_base.data_vars}
         self.set_grid(ds_base.rename(rmdict))
+        # Call basins once to set it
+        self.basins
```

Comment:
move this up

Reply:
Can't, it is derived from grid when not read from geoms. And the grid with the basins is first set in the line above where I call `self.basins`.